src/components/atoms/Timeline/index.tsx

```diff
@@ -73,33 +84,57 @@ const Timeline: React.FC<Props> = memo(
       return range;
     }, [_range]);
     const { start, end } = range;
+    const { zoom, events } = useTimelineInteraction({ range, onClick, onDrag });
+    const gapHorizontal = GAP_HORIZONTAL * (zoom - Math.trunc(zoom) + 1);
+    const memoryInterval = Math.max(
+      MEMORY_INTERVAL - Math.trunc(zoom - 1) * MEMORY_ZOOM_INTERVAL * MINUTES_SEC,
+      MINUTES_SEC,
+    );
+    const strongMemoryHours = MAX_ZOOM_RATIO - Math.trunc(zoom) + 1;
     const epochDiff = end - start;
 
     // convert epoch diff to second.
     const memoryCount = useMemo(
-      () => Math.trunc(epochDiff / EPOCH_SEC / MEMORY_INTERVAL),
-      [epochDiff],
+      () => Math.trunc(epochDiff / EPOCH_SEC / memoryInterval),
+      [epochDiff, memoryInterval],
     );
 
     const startDate = useMemo(() => new Date(start), [start]);
-    const hoursCount = Math.trunc(HOURS_SECS / MEMORY_INTERVAL);
+    const hoursCount = Math.trunc(HOURS_SECS / memoryInterval);
     // Convert memory count to pixel.
     const currentPosition = useMemo(() => {
-      const diff = Math.min((currentTime - start) / EPOCH_SEC / MEMORY_INTERVAL, memoryCount);
-      const strongMemoryCount = diff / (hoursCount * STRONG_MEMORY_HOURS);
+      const diff = Math.min((currentTime - start) / EPOCH_SEC / memoryInterval, memoryCount);
+      const strongMemoryCount = diff / (hoursCount * strongMemoryHours);
       return Math.max(
-        diff * GAP_HORIZONTAL +
+        diff * gapHorizontal +
           (diff - strongMemoryCount) * NORMAL_MEMORY_WIDTH +
           strongMemoryCount * STRONG_MEMORY_WIDTH +
           STRONG_MEMORY_WIDTH / 2,
         0,
       );
-    }, [currentTime, start, memoryCount, hoursCount]);
+    }, [
+      currentTime,
+      start,
+      memoryCount,
+      hoursCount,
+      gapHorizontal,
+      memoryInterval,
+      strongMemoryHours,
+    ]);
 
     return (
       <Container>
-        <MemoryBox>
-          <MemoryList start={startDate} memoryCount={memoryCount} hoursCount={hoursCount} />
+        <MemoryBox {...events}>
+          <MemoryContainer style={{ gap: `0 ${gapHorizontal}px` }}>
```

Comment:
If there wasn't a specific reason to use inline styles, can you move the gap to MemoryContainer's style declaration below?
You might already know, but if you don't, you can pass `gapHorizontal` as a regular prop here, and then below in the styled component declaration write like so:
```
const MemoryContainer = styled.div<{gapHorizontal: number}>`.....`
```

Reply:
`gapHorizontal` is too many changed so we should pass this to style props for improving performance.
https://emotion.sh/docs/best-practices#use-the-style-prop-for-dynamic-styles